### PR TITLE
 Convert dev lab to teardown and cleanup rather than rekick

### PR DESCRIPTION
**Pull request by stevelle**
Between builds the dev gate job will only run a teardown on containers
 and perform a minimum clean-up on the hosts. This should make developer
 gating jobs run faster.

Reviews desired by
@nrb @hughsaunders @jacobwagner and others

### CODE REVIEW

**Comment by nrb** [comment]
This looks good to my knowledge. There's more work to be done, obviously, but for now we can focus on the relatively simple task of tearing things down.

A question - have you run this on a lab cluster yet?

**Comment by stevelle** [comment]
No I haven't run it on a lab cluster yet, most of it has been run on contrived environments to test the individual changes.

**Comment by hughsaunders** [comment]
I'm going to run this against gating <a href="http://10.127.52.2:8000/admin/rolemapper/host/?cluster__id__exact=5">gating cluster 1</a> and see how it goes.

File: playbooks/dev-labs/clean.yml [hunk 1]
```diff
@@ -1,13 +1,6 @@
 ---
-- name: Rekick cluster nodes
-  gather_facts: False
+- name: Teardown cluster nodes
   hosts: "cluster{{ CLUSTER_NUMBER }}"
   tasks:
```
Comment (hughsaunders) [inline playbooks/dev-labs/clean.yml]: Add host-misc-cleanup role here?

File: playbooks/dev-labs/clean.yml [hunk 2]
```diff
@@ -1,13 +1,6 @@
...
-    - name: Rekick and wait for nodes to become available
-      delegate_to: localhost
-      rekick:
-        host_ip: "{{ ansible_ssh_host }}"
-        host_name: "{{ inventory_hostname }}"
-        kick_wait: 2400
-        kick_tries: 2
-        djeep_url: "http://10.127.52.2:8000/api"
+    - name: Teardown containers
+      include: teardown/teardown-all.yml
```
Comment (hughsaunders) [inline playbooks/dev-labs/clean.yml]: teardown-all is a playbook so can't be included as a task.

```
---
- include: path/to/playbook.yml
```
See teardown-all as an example of a playbook that includes another.

File: playbooks/dev-labs/clean.yml (hunk 1 above)
Comment (stevelle) [inline playbooks/dev-labs/clean.yml]: teardown-hosts applies that role. solving the teardown-all should take care of this, no?

Comment (hughsaunders) [inline playbooks/dev-labs/clean.yml]: Ah yes, sorry I missed that.

**Comment by stevelle** [comment]
Closing, will issue a new (squashed) pull request when I have an update fix.

**Pull request closed without merging**